components/calendar/src/types.rs

```diff
@@ -41,37 +51,99 @@ pub struct FormattableYear {
 
     /// The year number in the current era (usually 1-based).
     pub number: i32,
+}
 
-    /// The year in the current cycle for cyclic calendars (1-indexed)
-    /// can be set to `None` for non-cyclic calendars
-    ///
-    /// For chinese and dangi it will be
-    /// a number between 1 and 60, for hypothetical other calendars it may be something else.
-    pub cyclic: Option<NonZeroU8>,
+/// Representation of a year as needed for formatting
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[non_exhaustive]
+pub struct FormattableYear {
+    /// The "extended year", typically anchored with year 1 as the year 1 of either the most modern or
+    /// otherwise some "major" era for the calendar
+    pub extended_year: i32,
+    /// The rest of the details about the year.
+    pub kind: FormattableYearKind,
+}
 
-    /// The related ISO year. This is normally the ISO (proleptic Gregorian) year having the greatest
-    /// overlap with the calendar year. It is used in certain date formatting patterns.
+/// The type of year: Calendars like Chinese don't have an era and instead format with cyclic years.
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[allow(clippy::exhaustive_enums)] // Only two kinds of years
+pub enum FormattableYearKind {
+    /// An Era and a numeric year in that era.
+    EraYear(Era, i32),
```

Comment:
Suggestion: name these fields

```suggestion
    EraYear {
        era: Era,
        year_number: i32,
    },
```